sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -401,7 +401,9 @@ class SessionCatalog(
       val db = formatDatabaseName(name.database.getOrElse(currentDb))
       val table = formatTableName(name.table)
       val relation =
-        if (name.database.isDefined || !tempTables.contains(table)) {
+        if (db == "information_schema") {
+          tempTables(s"$db.$table")
```

Comment:
Ah, there is a reason. Catalyst can not see `InformationSchema`. `InformationSchema` is designed to be outside of catalyst.